packages/@aws-cdk/cdk/lib/context.ts

```diff
@@ -17,38 +19,48 @@ export class ContextProvider {
 
   private readonly stack: Stack;
 
-  constructor(private context: Construct) {
+  constructor(
+    private readonly context: Construct,
+    private readonly provider: string,
+    private readonly props: {[key: string]: any} = {}) {
     this.stack = Stack.find(context);
```

Comment:
I think we just go like this in the constructor:

```
this.props.account = this.stack.env.account;
this.props.region = this.stack.env.region;
```

And then in all other places we just use `this.props`.